### DISCUSSION TITLE
xmake 源码编译安装问题

### DISCUSSION THREAD

**Discussion by KMPrg** [xmake-io/xmake#3576] [Q&A] [answered]
根据官网教程，尝试了下源码编译安装，但发现没有 makefile 文件。
[源码编译安装]
git clone --recursive https://github.com/xmake-io/xmake.git
cd ./xmake
./configure
make
./scripts/get.sh __local__ __install_only__
source ~/.xmake/profile

进行到 make 的时候，提示缺少 makefile 文件，查看了一下的确没有，
所以需要自己写 makefile 文件么。
但是我还不会写makefile文件，所以源码编译安装并不是给新手用的么。
没别的意思哈，就是有点好奇，也可能是我没有理解这个过程？
make 这一步有什么其它意思？

**Comment by waruqi** [comment, marked as answer]
确定执行了 ./configure？应该会自动生成  makefile的

**Reply by KMPrg** [reply]
> 确定执行了 ./configure？应该会自动生成 makefile的
你好，可能是我添加了 --prefix 的原因？
 ./configure --prefix=/usr/local/xmake
这里不能指定路径吗？我不指定路径试试看，谢谢大佬~

**Reply by KMPrg** [reply]
> 确定执行了 ./configure？应该会自动生成 makefile的

我又执行了一次，我发现的确有问题，没注意到···
这是结果：
$ ./configure
checking for platform ... mingw
checking for architecture ... x86_64
checking for make ... ok
checking for toolchain ... no
toolchain not found!
但是我在源码文件的 xmake 文件里看到有 toolchains 文件，不知道这里为什么说没发现。

对了，我用的msys2，不清楚是不是和这有关系，也许需要在真的Linux系统里才不会有问题？

**Reply by KMPrg** [reply]
抱歉，我的问题，我好像没有用 mingw64 操作，我重新试了下，发现结果不一样了，也许能行。

**Reply by waruqi** [reply]
msys下 得用mingw 编译器

**Reply by KMPrg** [reply]
非常感谢，的确需要在 mingw下。
已经源码编译安装成功，不过步骤不太一样，用的是 make install。
因为我看官网指导里说：./scripts/get.sh local install_only是给xmake开发者调试源码用的，我应该用不上，不知道我理解的对不。

**Reply by waruqi** [reply]
`make install` 会装到 /usr 下，不支持 xmake update 自更新，一些功能受限，且需要 sudo 

`get.sh __local__ __instlal_only__` 会安装到 `~/.local` ，支持 xmake update ，且还会额外安装 shell tab 补全整合，功能最完整，也是推荐的安装方式

**Reply by KMPrg** [reply]
原来如此，非常感谢！之前对这部分确实有点迷惑~
